### PR TITLE
feat: support distributed `EXPLAIN ANALYZE`

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)

 - https://github.com/GreptimeTeam/greptimedb/issues/2374
 - https://github.com/GreptimeTeam/greptimedb/issues/3608

## What's changed and what's your intention?

Now `EXPLAIN ANALYZE` will work correctly under distributed mode. The format follows discussion here https://github.com/GreptimeTeam/greptimedb/issues/2374#issuecomment-1869394009

See `tests/cases/distributed/explain/analyze.result` for the example. Or here is one from my env:

```sql
explain analyze SELECT count(*) FROM system_metrics;

+-------+------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| stage | node | plan                                                                                                                                                                            |
+-------+------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| 0     | 0    |  MergeScanExec: peers=[4402341478400(1025, 0), ] metrics=[output_rows: 1, greptime_exec_read_cost: 0, ready_time: 6352264, finish_time: 7509279, first_consume_time: 7165836, ] |
|       |      |                                                                                                                                                                                 |
| 1     | 0    |  AggregateExec: mode=Final, gby=[], aggr=[COUNT(greptime.public.system_metrics.ts)] metrics=[output_rows: 1, elapsed_compute: 108029, ]                                         |
|       |      |   CoalescePartitionsExec metrics=[output_rows: 32, elapsed_compute: 83055, ]                                                                                                    |
|       |      |     AggregateExec: mode=Partial, gby=[], aggr=[COUNT(greptime.public.system_metrics.ts)] metrics=[output_rows: 32, elapsed_compute: 334913, ]                                   |
|       |      |       RepartitionExec: partitioning=RoundRobinBatch(32), input_partitions=1 metrics=[repart_time: 1, fetch_time: 441565, send_time: 30325, ]                                    |
|       |      |         StreamScanAdapter { stream: "<SendableRecordBatchStream>" } metrics=[output_rows: 3, mem_used: 24, ]                                                                    |
|       |      |                                                                                                                                                                                 |
|       |      | Total rows: 1                                                                                                                                                                   |
+-------+------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/common/recordbatch/src/adapter.rs [hunk 1]
```diff
@@ -276,28 +278,110 @@ impl Stream for RecordBatchStreamAdapter {
...
+        self.record_batch_metrics.plan_metrics.push(plan_metric);
+
+        self.current_level += 1;
+        Ok(true)
     }
 
-    for child in df_plan.children() {
-        collect_metrics(&child, result);
+    fn post_visit(&mut self, _plan: &dyn ExecutionPlan) -> std::result::Result<bool, Self::Error> {
+        // the last miuns will underflow
```
Comment (evenyag) [inline src/common/recordbatch/src/adapter.rs:335]: What does miun mean?

File: src/common/recordbatch/src/adapter.rs [hunk 2]
```diff
@@ -276,28 +278,110 @@ impl Stream for RecordBatchStreamAdapter {
...
-        });
+/// An [ExecutionPlanVisitor] to collect metrics from a [ExecutionPlan].
+#[derive(Default)]
+pub struct MetricCollector {
+    current_level: usize,
+    pub record_batch_metrics: RecordBatchMetrics,
+}
+
+impl ExecutionPlanVisitor for MetricCollector {
+    type Error = !;
```
Comment (evenyag) [inline src/common/recordbatch/src/adapter.rs:289]: Maybe we can use `()`?

**Review by evenyag — APPROVED** [review]
Exquisite Design

File: src/common/recordbatch/src/adapter.rs (hunk 2 above)
Comment (waynexia) [reply src/common/recordbatch/src/adapter.rs:289]: They are different. `!` means one can't return `Error` in the type system

Comment (evenyag) [reply src/common/recordbatch/src/adapter.rs:289]: Just save one feature `#![feature(never_type)]`.

File: src/query/src/analyze.rs [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+#[derive(Debug)]
+pub struct DistAnalyzeExec {
+    input: DfPhysicalPlanRef,
+    schema: SchemaRef,
+    properties: PlanProperties,
+}
+
+impl DistAnalyzeExec {
+    /// Create a new AnalyzeExec
```
Comment (fengjiachun) [inline src/query/src/analyze.rs:53]:
```suggestion
    /// Create a new DistAnalyzeExec
```

**Pull request merged**